src/framebuffer/draw.rs

```diff
@@ -161,18 +78,16 @@ impl<'a> framebuffer::FramebufferDraw for core::Framebuffer<'a> {
         }
     }
 
-    fn fill_circle(&mut self, pos: Point2<i32>, rad: u32, v: color) -> mxcfb_rect {
-        for current in { 1..rad + 1 } {
-            for (x, y) in
-                line_drawing::BresenhamCircle::new(pos.x as i32, pos.y as i32, current as i32)
-            {
-                self.write_pixel(
-                    Point2 {
-                        x: x as i32,
-                        y: y as i32,
-                    },
-                    v,
-                );
+    fn fill_circle(&mut self, pos: cgmath::Point2<i32>, rad: u32, v: color) -> mxcfb_rect {
+        let rad_square = (rad * rad) as i32;
```

Comment:
Much better than the previous hacky `for current_rad in { 1..rad + 1 } { draw_circle(.., current_rad) }` approach. I like it.

Reply:
Worth noting that this isn't quite the same. It can be solved by adding a floating point multiplication to make the rad*rad approximately halfway to the next radius, so the error is spread evenly rather than always being negative.